scripts/lib/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -229,12 +229,12 @@ def run_phase(self, success_change=False):  # pylint: disable=arguments-differ
                 self.run_indv(suffix = self._run_two_suffix)
                 self._case_two_custom_postrun_action()
 
-        # Compare results
-        # Case1 is the "main" case, and we need to do the comparisons from there
-        self._activate_case1()
-        self._link_to_case2_output()
+                # Compare results
+                # Case1 is the "main" case, and we need to do the comparisons from there
+                self._activate_case1()
```

Comment:
These lines need to be nested under the `if`, but I don't think they need to be nested under the `with` (and that's consistent with @jgfouca 's original implementation). See also #1858 

That said, I don't fully understand the subtleties here, so if you and @jgfouca think these lines should be nested under the `with`, then I'll accept that.